dot/sync/chain_sync_test.go

```diff
@@ -0,0 +1,1183 @@
+					EndBlockHash:  nil,
+					Direction:     network.Ascending,
+					Max:           &max128,
+				},
+				{
+					RequestedData: network.RequestedDataHeader + network.RequestedDataBody + network.RequestedDataJustification,
+					StartingBlock: *variadic.MustNewUint32OrHash(1 + maxResponseSize),
+					EndBlockHash:  nil,
+					Direction:     network.Ascending,
+					Max:           &max128,
+				},
+			},
+		},
+		{
+			w: &worker{
+				startNumber:  uintPtr(1),
+				targetNumber: uintPtr(10),
+				targetHash:   common.Hash{0xa},
+				direction:    network.Ascending,
+				requestData:  bootstrapRequestData,
+			},
+			expected: []*network.BlockRequestMessage{
+				{
+					RequestedData: bootstrapRequestData,
+					StartingBlock: *variadic.MustNewUint32OrHash(1),
+					EndBlockHash:  &(common.Hash{0xa}),
+					Direction:     network.Ascending,
+					Max:           &max128,
+				},
+			},
+		},
+		{
+			w: &worker{
+				startNumber:  uintPtr(1),
+				startHash:    common.Hash{0xb},
+				targetNumber: uintPtr(10),
+				targetHash:   common.Hash{0xc},
+				direction:    network.Ascending,
+				requestData:  bootstrapRequestData,
+			},
+			expected: []*network.BlockRequestMessage{
+				{
+					RequestedData: bootstrapRequestData,
+					StartingBlock: *variadic.MustNewUint32OrHash(common.Hash{0xb}),
+					EndBlockHash:  &(common.Hash{0xc}),
+					Direction:     network.Ascending,
+					Max:           &max128,
+				},
+			},
+		},
+		{
+			w: &worker{
+				startNumber:  uintPtr(10),
+				targetNumber: uintPtr(10),
+				direction:    network.Ascending,
+				requestData:  bootstrapRequestData,
+			},
+			expected: []*network.BlockRequestMessage{
+				{
+					RequestedData: bootstrapRequestData,
+					StartingBlock: *variadic.MustNewUint32OrHash(10),
+					Direction:     network.Ascending,
+					Max:           &max128,
+				},
+			},
+		},
+		{
+			w: &worker{
+				startNumber:  uintPtr(1 + maxResponseSize + (maxResponseSize / 2)),
+				targetNumber: uintPtr(1),
+				direction:    network.Descending,
+				requestData:  bootstrapRequestData,
+			},
+			expected: []*network.BlockRequestMessage{
+				{
+					RequestedData: network.RequestedDataHeader + network.RequestedDataBody + network.RequestedDataJustification,
+					StartingBlock: *variadic.MustNewUint32OrHash(1 + (maxResponseSize / 2)),
+					EndBlockHash:  nil,
+					Direction:     network.Descending,
+					Max:           &max64,
+				},
+				{
+					RequestedData: bootstrapRequestData,
+					StartingBlock: *variadic.MustNewUint32OrHash(1 + maxResponseSize + (maxResponseSize / 2)),
+					EndBlockHash:  nil,
+					Direction:     network.Descending,
+					Max:           &max128,
+				},
+			},
+		},
+	}
+
+	for i, tc := range testCases {
+		reqs, err := workerToRequests(tc.w)
+		require.NoError(t, err, fmt.Sprintf("case %d failed", i))
+		require.Equal(t, len(tc.expected), len(reqs), fmt.Sprintf("case %d failed", i))
+		require.Equal(t, tc.expected, reqs, fmt.Sprintf("case %d failed", i))
```

Comment:
- the test case index will be taken care of by the subtest name
- the len check is done by the last assertion

```suggestion
		assert.NoError(t, err)
		assert.Equal(t, tc.expected, reqs)
```

Reply:
Updated.